src/app/child-dev-project/children/child-select/child-select.component.ts

```diff
@@ -72,10 +75,18 @@ export class ChildSelectComponent implements OnChanges {
 
     this.suggestions = this.allChildren.filter((child) => {
       const key = "" + child.name + " " + child.projectNumber;
-      return key.toLowerCase().includes(this.searchText);
+      const isActive = child.isActive();
+      const matchesKey = key.toLowerCase().includes(this.searchText);
+      return this.showOnlyActiveChildren ? matchesKey && isActive : matchesKey;
```

Comment:
```suggestion
      if (this.showOnlyActiveChildren && !child.isActive()) {
        return false;
      }
      return key.toLowerCase().includes(this.searchText);
```

Having the two conditions separate makes it a bit easier to read in my opinion.